### PR TITLE
Added transform 2d header and sample

### PR DESCRIPTION
**Pull request by Smiley98**
I've made a sample project to illustrate the added functionality. Please let me know if I need to make any changes!

### CODE REVIEW

File: transform_2d.h [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+/*******************************************************************************************
+*
+*   2d transformations
+*
+*   Copyright (c) 2022 Jeffery Myers
+*
+********************************************************************************************/
+
+
+// define RAY2D_TRANSFORM_IMPLEMENTATION in one and only one c/cpp files
```
Comment (JeffM2501) [inline transform_2d.h:11]: this does not match the define used below
TRANSFORM2D_IMPLEMENTATION

File: transform_2d.h [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+		Matrix transformation = MatrixMultiply(MatrixMultiply(scale, rotation), translation);
+		for (int i = 0; i < pointCount; i++)
+			points[i] = Vector2Transform(points[i], transformation);
+	}
+
+#endif //TRANSFORM2D_IMPLEMENTATION
+
+#if defined(__cplusplus)
+}
+#endif
```
Comment (JeffM2501) [inline transform_2d.h:62]: files need to end in a newline

File: transform_2d/samples/example.c [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        DrawText("SPACE / LSHIFT to scale up/down", 10, 10, 20, RED);
+        DrawText("W / S to move forwards/backwards", 10, 30, 20, ORANGE);
+        DrawText("A / D to move left/right", 10, 50, 20, BLUE);
+        DrawText("E / Q to rotate clockwise/counter-clockwise", 10, 70, 20, PURPLE);
+        DrawText("Mouse-over the polygon to change its color", 10, 90, 20, GREEN);
+        EndDrawing();
+    }
+
+	CloseWindow();
+	return 0;
```
Comment (JeffM2501) [inline transform_2d/samples/example.c:115]: spacing is incorrect

File: transform_2d/samples/example.c [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        DrawCircleV(mouse, 5.0f, DARKGRAY);
+        DrawCircleV(origin, 5.0f, DARKGRAY);
+        DrawLineEx(origin, Vector2Add(origin, Vector2Scale(forward, transform.scale)), 5.0f, DARKGRAY);
+        DrawLineStrip(points, sizeof points / sizeof points[0], color);
+        DrawText("SPACE / LSHIFT to scale up/down", 10, 10, 20, RED);
+        DrawText("W / S to move forwards/backwards", 10, 30, 20, ORANGE);
+        DrawText("A / D to move left/right", 10, 50, 20, BLUE);
+        DrawText("E / Q to rotate clockwise/counter-clockwise", 10, 70, 20, PURPLE);
+        DrawText("Mouse-over the polygon to change its color", 10, 90, 20, GREEN);
+        EndDrawing();
```
Comment (JeffM2501) [inline transform_2d/samples/example.c:111]: Please add some newlines here to space out the code, it's not clear what is happening

**Review by JeffM2501 — CHANGES_REQUESTED** [review]
I'm not sure how useful this is to just transform points. I think this should have functions to set the GL matrix and push and pop, just like the C++ 3d object transform has.

The utility of this would be improved by making it support nested transforms too.

**Comment by Smiley98** [comment]
Thanks for the feedback, I'll take a look at the matrix stack in the context of 2d and run my proposed implementation by you.

**Pull request closed without merging**